python-avd/pyavd/_eos_designs/structured_config/underlay/ethernet_interfaces.py

```diff
@@ -94,6 +94,8 @@ def ethernet_interfaces(self: AvdStructuredConfigUnderlay) -> list | None:
                 if link.get("ip_address") is not None:
                     if "unnumbered" in link["ip_address"].lower():
                         ethernet_interface["ip_address"] = link["ip_address"]
+                    elif self.shared_utils.underlay_ipv6:
+                        ethernet_interface["ipv6_address"] = f"{link['ip_address']}/{link['prefix_length']}"
```

Comment:
Notice that underlay config will be generated by the child, but will also be applied on the parent. So the parent may not have underlay_ipv6 set (Also underlay_ipv6 would not be enough in this case to avoid overlap with 5549).
So you probably need some error handling here to check for the ipv6 address.

Reply:
Also as mentioned elsewhere I would prefer not to reuse the ip_address field for both address families.